src/test/Base32Test.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+* Copyright (c) 2003-2023 Rony Shapiro <ronys@pwsafe.org>.
+* All rights reserved. Use of the code is allowed under the
+* Artistic License 2.0 terms, as specified in the LICENSE file
+* distributed with this code, or available from
+* http://www.opensource.org/licenses/artistic-license-2.0.php
+*/
+// TOTPTest.cpp: Unit test for TOTP.
+
+#ifdef WIN32
+#include "../ui/Windows/stdafx.h"
+#endif
+
+#include <algorithm>
+#include <vector>
+
+#include "core/crypto/external/Chromium/base32.h"
+#include "core/crypto/totp.h"
+#include "gtest/gtest.h"
+
+using namespace std;
+
+// The odd string splits below are an effort to mitigate
+// CI/CD false positives around secrets detection.
+vector<tuple<const char*, const char*>> base32_test_cases{
```

Comment:
What's the source of these test vectors? Worth stating, perhaps?

Reply:
PR updated w/test vectors including both non-printable and RFC4648 Test Vectors.